### DISCUSSION TITLE
Trouble running the test

### DISCUSSION THREAD

**Discussion by nataliyah123** [MDAnalysis/mdanalysis#4420] [GSoC Q&A] [answered]
hi @RMeli 
I went through the material you provided in the earlier [discussion](https://www.mdanalysis.org/pages/learning_MDAnalysis/#the-universe-as-balls-and-springs-molecular-dynamics-in-python). I ran the examples provided in the notebooks in my Jupyter and it worked fine. I tried to install MDanalysis on my WSL according to the instructions provided [here](https://userguide.mdanalysis.org/2.8.0-dev0/testing.html), I can import mdanalysis in WSL terminal successfully, my version is 2.8.0.dev0, but when I try to run test using the command pytest  --disable-pytest-warnings , I get 155 errors, saying the same thing  ModuleNotFoundError: No module named 'MDAnalysis'.

**Comment by RMeli** [comment, marked as answer]
H @nataliyah123, this is unfortunately not enough information to help you with this issue. Which version of Python are you using? How did you install Python and MDAnalysis?

If you installed MDAnalysis and see you have version `2.8.0.dev0` and have `ModuleNotFoundError: No module named 'MDAnalysis'` errors when running `pytest`, the first thing that comes to mind is that `pytest` is not not running with the same interpreter you used to install MDAnalysis. I'd use `python -m pytest ...` to run `pytest` (you can check if `python` links to the right interpreter with `which python`).

**Reply by nataliyah123** [reply]
Thanks for the reply. your solution did resolve the following issue.
![image](https://github.com/MDAnalysis/mdanalysis/assets/5617255/f9fd5992-03aa-499a-9040-9c328a4dc571)
this I am pretty sure is caused by me because I tried !pip install --upgrade MDAnalysisTests on jupyter (just trying to test the release version, I removed it).  my python version is 3.12.1.  The instructions provided [here](https://userguide.mdanalysis.org/stable/contributing_code.html) is how I installed mdanalysis. my mdanalysis version is 2.8.0-dev0

**Reply by RMeli** [reply]
You need to make sure that `mdanalysis` and `mdanalysistests` versions match. If your version of MDAnalysis is `2.8.0-dev0`, it means you have a development version installed manually. Therefore, you need to also install manually `mdanalysistests`, and not just `pip install mdanalysistests`. The latter command will download and install the latest _released_ version (i.e. `2.7.0`) from PyPI.

The [Contributing to the main codebase](https://userguide.mdanalysis.org/stable/contributing_code.html) page you linked shows you how to install `mdanalysis` and `mdanalysistests` from source (once you have all other dependencies installed):

```bash
# go to the mdanalysis source directory
cd mdanalysis/

# Build and install the MDAnalysis package
cd package/
python -m pip install -e .

# Build and install the test suite
cd ../testsuite/
python -m pip install -e .
```

You need to do the last step, instead of just doing `pip install --upgrade mdanalysistests`.

---

Side node: please try not to share images of code, but copy-paste the code into a [code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) directly. Images are not searchable, while text/code is.

**Reply by nataliyah123** [reply]
I followed the above tutorial and I checked that both mdanalysis and mdanalysistests were in my conda list in my env. but i was getting `modulenotfound` error, (do you want me to paste the output in code block also or just the code, I have used the code block below, let me know if this is how you want it. )
![mda_on_wsl](https://github.com/MDAnalysis/mdanalysis/assets/5617255/4e5c5683-a31b-4530-b835-ebe5050d54c2)
I thought that it was some sort of path error. so I tried different things. One of the thing was to see what` !pip install --upgrade MDAnalysisTests` was doing from jupyter.  

I got Defaulting to user installation because normal site-packages is not writeable and the following output. 

Defaulting to user installation because normal site-packages is not writeable
```
Collecting MDAnalysisTests
  Using cached MDAnalysisTests-2.7.0-py3-none-any.whl
Requirement already satisfied: hypothesis in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysisTests) (6.92.3)
Requirement already satisfied: pytest>=3.3.0 in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysisTests) (7.4.4)
Collecting MDAnalysis==2.7.0
  Using cached MDAnalysis-2.7.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (10.1 MB)
Requirement already satisfied: mda-xdrlib in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (0.2.0)
Requirement already satisfied: fasteners in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (0.19)
Requirement already satisfied: tqdm>=4.43.0 in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (4.66.1)
Requirement already satisfied: numpy<2.0,>=1.22.3 in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (1.26.3)
Requirement already satisfied: packaging in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (23.2)
Requirement already satisfied: scipy>=1.5.0 in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (1.11.4)
Requirement already satisfied: matplotlib>=1.5.1 in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (3.8.2)
Requirement already satisfied: joblib>=0.12 in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (1.3.2)
Requirement already satisfied: GridDataFormats>=0.4.0 in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (1.0.2)
Requirement already satisfied: threadpoolctl in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (3.2.0)
Requirement already satisfied: mmtf-python>=1.0.0 in /home/ibia/.local/lib/python3.10/site-packages (from MDAnalysis==2.7.0->MDAnalysisTests) (1.1.3)
Requirement already satisfied: exceptiongroup>=1.0.0rc8 in /home/ibia/.local/lib/python3.10/site-packages (from pytest>=3.3.0->MDAnalysisTests) (1.2.0)
Requirement already satisfied: iniconfig in /home/ibia/.local/lib/python3.10/site-packages (from pytest>=3.3.0->MDAnalysisTests) (2.0.0)
Requirement already satisfied: pluggy<2.0,>=0.12 in /home/ibia/.local/lib/python3.10/site-packages (from pytest>=3.3.0->MDAnalysisTests) (1.3.0)
Requirement already satisfied: tomli>=1.0.0 in /home/ibia/.local/lib/python3.10/site-packages (from pytest>=3.3.0->MDAnalysisTests) (2.0.1)
Requirement already satisfied: attrs>=22.2.0 in /home/ibia/.local/lib/python3.10/site-packages (from hypothesis->MDAnalysisTests) (23.2.0)
Requirement already satisfied: sortedcontainers<3.0.0,>=2.1.0 in /home/ibia/.local/lib/python3.10/site-packages (from hypothesis->MDAnalysisTests) (2.4.0)
Requirement already satisfied: mrcfile in /home/ibia/.local/lib/python3.10/site-packages (from GridDataFormats>=0.4.0->MDAnalysis==2.7.0->MDAnalysisTests) (1.4.3)
Requirement already satisfied: kiwisolver>=1.3.1 in /home/ibia/.local/lib/python3.10/site-packages (from matplotlib>=1.5.1->MDAnalysis==2.7.0->MDAnalysisTests) (1.4.5)
Requirement already satisfied: cycler>=0.10 in /home/ibia/.local/lib/python3.10/site-packages (from matplotlib>=1.5.1->MDAnalysis==2.7.0->MDAnalysisTests) (0.12.1)
Requirement already satisfied: fonttools>=4.22.0 in /home/ibia/.local/lib/python3.10/site-packages (from matplotlib>=1.5.1->MDAnalysis==2.7.0->MDAnalysisTests) (4.47.0)
Requirement already satisfied: pillow>=8 in /home/ibia/.local/lib/python3.10/site-packages (from matplotlib>=1.5.1->MDAnalysis==2.7.0->MDAnalysisTests) (10.2.0)
Requirement already satisfied: python-dateutil>=2.7 in /home/ibia/.local/lib/python3.10/site-packages (from matplotlib>=1.5.1->MDAnalysis==2.7.0->MDAnalysisTests) (2.8.2)
Requirement already satisfied: pyparsing>=2.3.1 in /usr/lib/python3/dist-packages (from matplotlib>=1.5.1->MDAnalysis==2.7.0->MDAnalysisTests) (2.4.7)
Requirement already satisfied: contourpy>=1.0.1 in /home/ibia/.local/lib/python3.10/site-packages (from matplotlib>=1.5.1->MDAnalysis==2.7.0->MDAnalysisTests) (1.2.0)
Requirement already satisfied: msgpack>=1.0.0 in /home/ibia/.local/lib/python3.10/site-packages (from mmtf-python>=1.0.0->MDAnalysis==2.7.0->MDAnalysisTests) (1.0.7)
Requirement already satisfied: six>=1.5 in /usr/lib/python3/dist-packages (from python-dateutil>=2.7->matplotlib>=1.5.1->MDAnalysis==2.7.0->MDAnalysisTests) (1.16.0)
Installing collected packages: MDAnalysis, MDAnalysisTests
Successfully installed MDAnalysis-2.7.0 MDAnalysisTests-2.7.0
```

 I think the pip has installed mdanalysis and mdanalysistest outside of my env. because my env address is /home/ibia/anaconda3/envs/mdanalysis-dev and pip has installed in /home/ibia/.local/lib/python3.10/site-packages , after this the module not found error was gone. but the test had errors, when your reply came, I used python -m pytest and the errors were gone because now python version 3.12 was used instead of 3.10.  

Now when i try to use `make doctest` I receive the following error 
![make_doctest](https://github.com/MDAnalysis/mdanalysis/assets/5617255/14a19819-635d-4542-87e2-e858afc76791)
python310 instead of active python312 is used

**Reply by RMeli** [reply]
> I followed the above tutorial and I checked that both mdanalysis and mdanalysistests were in my conda list in my env. but i was getting modulenotfound error

From your Jupyter Notebook output, looks like the Python interpreter used by `pytest` is version 3.10.12, which is different from the version of the interpreter of your active environment (3.12.1). This points to issues with your installation/environment. I think the best way would be to start from a fresh environment (make sure you activate it immediately after creating it!) and follow the User Guide step by step.

> do you want me to paste the output in code block also or just the code

I did not realise you were using Jupyter Notebooks from your previous message, I thought it was a screenshot of a terminal window. I have reservations on Jupyter Notebooks (see the excellent [I don't like notebooks](https://www.youtube.com/watch?v=7jiPeIFXb6U) talk by Joel Grus), and in this case it might actually be better to post a screenshot.

**Reply by nataliyah123** [reply]
`python -m pip install msmb_theme==1.2.0` is throwing the follwing error. 
```
Collecting msmb_theme
  Using cached msmb_theme-1.2.0.tar.gz (54 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      /tmp/pip-install-6g0ik8_f/msmb-theme_ce47477fd93e4fdfa525b8b1542102c4/versioneer.py:467: SyntaxWarning: invalid escape sequence '\s'
        LONG_VERSION_PY['git'] = '''
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-6g0ik8_f/msmb-theme_ce47477fd93e4fdfa525b8b1542102c4/setup.py", line 15, in <module>
          version=versioneer.get_version(),
                  ^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-install-6g0ik8_f/msmb-theme_ce47477fd93e4fdfa525b8b1542102c4/versioneer.py", line 1405, in get_version
          return get_versions()["version"]
                 ^^^^^^^^^^^^^^
        File "/tmp/pip-install-6g0ik8_f/msmb-theme_ce47477fd93e4fdfa525b8b1542102c4/versioneer.py", line 1339, in get_versions
          cfg = get_config_from_root(root)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-install-6g0ik8_f/msmb-theme_ce47477fd93e4fdfa525b8b1542102c4/versioneer.py", line 399, in get_config_from_root
          parser = configparser.SafeConfigParser()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
(mdanalysis-dev) ibia@Sumaira:/mnt/c/WINDOWS/system32$ python -m pip install --upgrade msmb_theme
Collecting msmb_theme
  Using cached msmb_theme-1.2.0.tar.gz (54 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      /tmp/pip-install-qeb30rrt/msmb-theme_c4b84e1edd104d35b0ab70ff01694d98/versioneer.py:467: SyntaxWarning: invalid escape sequence '\s'
        LONG_VERSION_PY['git'] = '''
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-qeb30rrt/msmb-theme_c4b84e1edd104d35b0ab70ff01694d98/setup.py", line 15, in <module>
          version=versioneer.get_version(),
                  ^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-install-qeb30rrt/msmb-theme_c4b84e1edd104d35b0ab70ff01694d98/versioneer.py", line 1405, in get_version
          return get_versions()["version"]
                 ^^^^^^^^^^^^^^
        File "/tmp/pip-install-qeb30rrt/msmb-theme_c4b84e1edd104d35b0ab70ff01694d98/versioneer.py", line 1339, in get_versions
          cfg = get_config_from_root(root)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-install-qeb30rrt/msmb-theme_c4b84e1edd104d35b0ab70ff01694d98/versioneer.py", line 399, in get_config_from_root
          parser = configparser.SafeConfigParser()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details. 
```

I tried `python -m pip install --upgrade msmb_theme` got the same error. Can I replace instances of configparser.SafeConfigParser with configparser.RawConfigParser. I am already in my environment and my python version is 3.12.1

**Reply by RMeli** [reply]
I think that `versioneer.py`, at least the version used in `msmb_theme` (which hasn't been updated since 2016), is too old and not supported by the latest versions of Python.

This was the first hit on a Google search of the error: https://github.com/pydata/pandas-datareader/issues/969#issuecomment-1786843855.

What are you trying to achieve with `msmb_theme`?

**Comment by nataliyah123** [comment]
![image](https://github.com/MDAnalysis/mdanalysis/assets/5617255/99134ae4-af7c-4726-ae97-b32dc9a313fa)

Just trying to folllow the tutorial step by step
`# documentation dependencies
conda install -c conda-forge sphinx pybtex pybtex-docutils \
    sphinxcontrib-bibtex sphinx_rtd_theme sphinx-sitemap
python -m pip install msmb_theme==1.2.0`

msmb_theme library applies slight modifications to the sphinx_rtd_theme.

**Reply by nataliyah123** [reply]
skipped the msmb_theme installation and followed the rest, everything is working including pytest, except make doctest, got the following error
```
sphinx-build -v -W -b doctest   source ../doctest
Running Sphinx v7.2.6
Original exception:
Traceback (most recent call last):
  File "/home/ibia/anaconda3/envs/mdanalysis-dev/lib/python3.12/site-packages/sphinx/registry.py", line 447, in load_extension
    mod = import_module(extname)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ibia/anaconda3/envs/mdanalysis-dev/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'mdanalysis_sphinx_theme'


Traceback (most recent call last):
  File "/home/ibia/anaconda3/envs/mdanalysis-dev/lib/python3.12/site-packages/sphinx/registry.py", line 447, in load_extension
    mod = import_module(extname)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ibia/anaconda3/envs/mdanalysis-dev/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'mdanalysis_sphinx_theme'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/ibia/anaconda3/envs/mdanalysis-dev/lib/python3.12/site-packages/sphinx/cmd/build.py", line 293, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ibia/anaconda3/envs/mdanalysis-dev/lib/python3.12/site-packages/sphinx/application.py", line 233, in __init__
    self.setup_extension(extension)
  File "/home/ibia/anaconda3/envs/mdanalysis-dev/lib/python3.12/site-packages/sphinx/application.py", line 406, in setup_extension
    self.registry.load_extension(self, extname)
  File "/home/ibia/anaconda3/envs/mdanalysis-dev/lib/python3.12/site-packages/sphinx/registry.py", line 450, in load_extension
    raise ExtensionError(__('Could not import extension %s') % extname,
sphinx.errors.ExtensionError: Could not import extension mdanalysis_sphinx_theme (exception: No module named 'mdanalysis_sphinx_theme')

Extension error:
Could not import extension mdanalysis_sphinx_theme (exception: No module named 'mdanalysis_sphinx_theme')
make: *** [Makefile:130: doctest] Error 2  
```
python 3.12 has issues with mdanalysis dependencies. I have two options either I downgrade to a lower python version or just make the changes I mentioned in the previous post

**Comment by nataliyah123** [comment]
here is further info about this closed [issue](https://github.com/MDAnalysis/mdanalysis/issues/4441). just in case if someone else runs into this issue.
